### DISCUSSION TITLE
corner probing and z probing make program handle

### DISCUSSION THREAD

**Discussion by wared** [terjeio/ioSender#309] [General]
i just want to report some bug i encounter, while auto probing one corner and z the program like freeze but i still have the enabling keyboard jogging option available and button reacts as i passe on them but at the moment coordinates should update on the DRO it just handle

my specs : win10, GRBLAHL2K, iosender 2.0.41 (i am updrading atm)

it happend twice and seems to be reproductible, i will tell if the software upgrade resolve it

**Comment by wared** [comment]
while digging into i found what append

i leave the touchplate height empty, with no value instead of putting "0"

**Reply by terjeio** [reply]
I'll fix this in the next release, `NaN` (Not A Number) was returned from the field when empty,  should return 0.

**Comment by wared** [comment]
hi terjeio,

firstely, thanks for your work i really enjoy this tool

maybe it has been resolved in the update too but i have some other strange behavior, when machining, the speed change i could ask via the feed speed control buttons sometimes take time to show or i just can't change it at all when the job is running, when machine is in idle state i can modify it as i want with instant change in the software

i hope it is clear, as i am not english native my explainations can be tricky to understand

**Reply by terjeio** [reply]
> , the speed change i could ask via the feed speed control buttons sometimes take time to show or i just can't change it at all when the job is running

Do you have more details as I am not able to replicate this.
If you select _Help > About > To clipboard_ and paste in a comment that may help.
Is the gcode a lot of short moves?
Both RPM (speed) and feed rate is affected?

**Reply by wared** [reply]
thanks for your repluy, here is the "help>about" : 

%
; grblHAL
; 1.1f.20220710
; [OPT:VNMZSL2,191,1024,3,0]
; [NEWOPT:ENUMS,RT+,HOME,ES,TC,SED,ETH]
; [FIRMWARE:grblHAL]
; [NVS STORAGE:*FLASH]
; [DRIVER:iMXRT1062]
; [DRIVER VERSION:220710]
; [DRIVER OPTIONS:USB.1]
; [BOARD:GRBLHAL2000 - PRINTNC]
; [AUX IO:4,1,0,0]
; [IP:192.168.0.124]
; [NETCON:Telnet]
; [PLUGIN:MODBUS v0.13]
; [PLUGIN:VFD SELECTOR v0.02]
; [PLUGIN:HUANYANG v1 v0.07]
; [PLUGIN:KEYPAD v1.3 INTERTEST]
; [PLUGIN:Macro plugin v0.02]
; [PLUGIN:RGB Indicator Lights v1.0]
;
$N0=
$N1=
; 0 - Step pulse time
$0=5.0
; 1 - Step idle delay
$1=255
; 2 - Step pulse invert
$2=0
; 3 - Step direction invert
$3=7
; 4 - Invert stepper enable pin(s)
$4=7
; 5 - Invert limit pins
$5=0
; 6 - Invert probe pin
$6=1
; 8 - Ganged axes direction invert
$8=0
; 9 - PWM Spindle
$9=1
; 10 - Status report options
$10=511
; 11 - Junction deviation
$11=0.010
; 12 - Arc tolerance
$12=0.002
; 13 - Report in inches
$13=0
; 14 - Invert control pins
$14=70
; 15 - Invert coolant pins
$15=0
; 16 - Invert spindle signals
$16=0
; 17 - Pullup disable control pins
$17=0
; 18 - Pullup disable limit pins
$18=0
; 19 - Pullup disable probe pin
$19=0
; 20 - Soft limits enable
$20=1
; 21 - Hard limits enable
$21=1
; 22 - Homing cycle
$22=9
; 23 - Homing direction invert
$23=1
; 24 - Homing locate feed rate
$24=50.0
; 25 - Homing search seek rate
$25=1500.0
; 26 - Homing switch debounce delay
$26=250
; 27 - Homing switch pull-off distance
$27=1.000
; 28 - G73 Retract distance
$28=0.100
; 29 - Pulse delay
$29=0.0
; 30 - Maximum spindle speed
$30=1000.000
; 31 - Minimum spindle speed
$31=0.000
; 32 - Mode of operation
$32=0
; 33 - Spindle PWM frequency
$33=5000
; 34 - Spindle PWM off value
$34=0.0
; 35 - Spindle PWM min value
$35=0.0
; 36 - Spindle PWM max value
$36=100.0
; 37 - Steppers deenergize
$37=0
; 39 - Enable legacy RT commands
$39=1
; 40 - Limit jog commands
$40=1
; 43 - Homing passes
$43=1
; 44 - Axes homing, first pass
$44=4
; 45 - Axes homing, second pass
$45=3
; 46 - Axes homing, third pass
$46=0
; 50 - Step jog speed
$50=100.0
; 51 - Slow jog speed
$51=600.0
; 52 - Fast jog speed
$52=3000.0
; 53 - Step jog distance
$53=0.250
; 54 - Slow jog distance
$54=500.0
; 55 - Fast jog distance
$55=3000.0
; 62 - Sleep enable
$62=0
; 63 - Feed hold actions
$63=2
; 64 - Force init alarm
$64=0
; 65 - Probing feed override
$65=0
; 70 - Network Services
$70=3
; 100 - X-axis travel resolution
$100=99.948
; 101 - Y-axis travel resolution
$101=100.114
; 102 - Z-axis travel resolution
$102=199.854
; 110 - X-axis maximum rate
$110=7000.000
; 111 - Y-axis maximum rate
$111=7000.000
; 112 - Z-axis maximum rate
$112=1000.000
; 120 - X-axis acceleration
$120=1000.000
; 121 - Y-axis acceleration
$121=1000.000
; 122 - Z-axis acceleration
$122=10.000
; 130 - X-axis maximum travel
$130=1010.000
; 131 - Y-axis maximum travel
$131=980.000
; 132 - Z-axis maximum travel
$132=140.000
; 160 - X-axis backlash compensation
$160=0.000
; 161 - Y-axis backlash compensation
$161=0.000
; 162 - Z-axis backlash compensation
$162=0.000
; 170 - X-axis dual axis offset
$170=0.000
; 171 - Y-axis dual axis offset
$171=0.000
; 172 - Z-axis dual axis offset
$172=0.000
; 300 - Hostname
$300=GRBLHAL2000
; 301 - IP Mode
$301=0
; 302 - IP Address
$302=192.168.0.124
; 303 - Gateway
$303=192.168.0.254
; 304 - Netmask
$304=255.255.255.0
; 305 - Telnet port
$305=23
; 307 - Websocket port
$307=80
; 341 - Tool change mode
$341=0
; 342 - Tool change probing distance
$342=30.0
; 343 - Tool change locate feed rate
$343=25.0
; 344 - Tool change search seek rate
$344=200.0
; 345 - Tool change probe pull-off rate
$345=100.0
; 347 - Dual axis length fail
$347=5.0
; 348 - Dual axis length fail min
$348=2.500
; 349 - Dual axis length fail max
$349=10.000
; 370 - Invert I/O Port inputs
$370=0
; 372 - Invert I/O Port outputs
$372=0
; 374 - ModBus baud rate
$374=3
; 375 - ModBus RX timeout
$375=50
; 384 - Disable G92 persistence
$384=0
; 395 - Default spindle
$395=0
; 450 - Macro 1 UP
$450=
; 451 - Macro 2 RIGHT
$451=
; 452 - Macro 3 DOWN
$452=
; 453 - Macro 4 LEFT
$453=
; 454 - Macro 5 SPINDLE
$454=S200M03
; 455 - Macro 6 RAISE
$455=
; 456 - Macro 7 LOWER
$456=
; 461 - RPM per Hz
$461=60
; 462 - Run/Stop Register (decimal)
$462=8192
; 463 - Set Frequency Register (decimal)
$463=8193
; 464 - Get Frequency Register (decimal)
$464=8451
; 465 - Run CW Command (decimal)
$465=18
; 466 - Run CCW Command (decimal)
$466=34
; 467 - Stop Command (decimal)
$467=1
; 468 - RPM input Multiplier
$468=50
; 469 - RPM input Divider
$469=60
; 470 - RPM output Multiplier
$470=60
; 471 - RPM output Divider
$471=100
; 473 - VFD Model
$473=0
%

computer is a i9 4th gen with 16gb RAM, ssd, fresh windows10 install, updated and unplugged from the network as it is dedicated to the cnc, i tranfer files with bluetooth. i did not notice any usage peak in the task manager while using it

it looks like a lag, sometimes before runing a cycle i change the feed rate from 100% to 70%, so i just click three times and do other things, i see the software drops to 80%, hangs for a second or two and goes to 70%

my spindle is not connected to my grblhal board so i only use the feed change, and as it does not respond as it should i tend to not use it at all with time... 

let me know if i can help you in any way to try to understand what append

**Reply by terjeio** [reply]
I suggest you build (and reflash) an up to date version of the firmware with the [Web Builder](http://svn.io-engineering.com:8080/?driver=iMXRT1062&board=GRBLHAL2000%20-%20PRINTNC) without unused plugins/options and try that.
Backup/restore your controller settings via buttons in the _Settings:Grbl_ tab. You will get a bunch of error messages on restore due to setting options for plugins not beeing available - those can be ignored. Note that restore has to be done via an USB connection as you are connecting via ethernet (Telnet) since the settings for ethernet will be reset on reflashing.
